datasets/vggface2.py

```diff
@@ -10,80 +10,425 @@
+        nf_number = 0
+        n_words = 0
+        pickle_dict = {key: [] for key in ["boxes", "landmarks", "img_list", "lbl_list"]}
+        pickle_dict["word2index"] = {}
 
-        t_start = time.time()
-        print('Data loading...')
-        for n_file, data_file in enumerate(data_file_list):
-            if ((n_file+1) % 5) == 0:
-                print(f'\t{n_file+1} of {len(data_file_list)}')
-            f_path = os.path.join(data_folder, data_file)
+        for jpg in tqdm(img_paths):
+            boxes = []
+            image = cv2.imread(jpg)
+            bboxes, lndmrks = retina.detect(image)
+            if len(bboxes) == 0:
+                nf_number += 1
+                continue
 
-            with open(f_path, 'rb') as f:
-                x = pickle.load(f)
+            for box in bboxes:
+                box = np.clip(box[:4], 0, None)
+                boxes.append(box)
 
-            for key in list(x)[::resample_subj]:
-                val = x[key]
-                for key2 in list(val)[::resample_img_per_subj]:
-                    self.img_list.append(val[key2]['img'])
-                    self.embedding_list.append(np.array(val[key2]['embedding']).astype(np.float32))
-                    self.sid_list.append(subj_idx)
-                    n_elems += 1
-                subj_idx += resample_subj
+            lndmrks = lndmrks[0]
 
-        t_end = time.time()
-        print(f'{n_elems} of data samples loaded in {t_end-t_start:.4f} seconds.')
+            dir_name = os.path.dirname(jpg)
+            lbl = os.path.relpath(dir_name, self.d_path)
 
-    def __normalize_data(self, data_item):
-        data_item = data_item.astype(np.float32)
-        data_item /= 256
-        return data_item
+            if lbl not in pickle_dict["word2index"]:
+                pickle_dict["word2index"][lbl] = n_words
+                n_words += 1
+
+            pickle_dict["lbl_list"].append(lbl)
+            pickle_dict["boxes"].append(boxes)
+            pickle_dict["landmarks"].append(lndmrks)
+            pickle_dict["img_list"].append(os.path.relpath(jpg, self.dataset_path))
+        if nf_number > 0:
+            print(f'Not found any faces in {nf_number} images ')
+
+        with open(self.gt_path, 'wb') as f:
+            pickle.dump(pickle_dict, f)
 
     def __len__(self):
-        return len(self.img_list)
+        return len(self.pickle_dict["img_list"]) - 1
+
+    def __getitem__(self, index):
+        if index >= len(self):
+            raise IndexError
+
+        if self.mode == 'detection':
+            return self.__getitem_detection(index)
+
+        if self.mode == 'identification':
+            return self.__getitem_identification(index)
+
+        if self.mode == 'identification_dr':
+            return self.__getitem_identification_dr(index)
+
+        # Will never reached
+        return None
+
+    def __getitem_detection(self, index):
+
```

Comment:
Please add a description comment for the function.

Reply:
Realized in the latest commit.